3_summarize/src/annual_thermal_metrics.R

```diff
@@ -0,0 +1,256 @@
+        # sthermo_depth_mean = sthermo_depth_mean(),
+        
+        peak_temp = peak_temp(date, wtr, depth),
+        gdd_wtr_0c = gdd_wtr_0c(date, wtr),
+        gdd_wtr_5c = gdd_wtr_5c(date, wtr),
+        gdd_wtr_10c = gdd_wtr_10c(date, wtr),
+        
+        # bottom_temp_at_strat = bottom_temp_at_strat(),
+        # schmidt_daily_annual_sum = schmidt_daily_annual_sum(),
+        
+        mean_surf_jas = mean_surf_jas(date, wtr, depth),
+        max_surf_jas = max_surf_jas(date, wtr, depth),
+        mean_bot_jas = mean_bot_jas(date, wtr, depth),
+        max_bot_jas = max_bot_jas(date, wtr, depth),
+        
+        # Per month calcs return a data.frame and get "unpacked" below 
+        mean_surf_mon = mean_surf_mon(date, wtr, depth),
+        max_surf_mon = max_surf_mon(date, wtr, depth),
+        mean_bot_mon = mean_bot_mon(date, wtr, depth),
+        max_bot_mon = max_bot_mon(date, wtr, depth),
+        
+        .groups = "keep" # suppresses message about regrouping
+      ) %>% 
+      unpack(cols = c(mean_surf_mon, max_surf_mon, 
+                      mean_bot_mon, max_bot_mon))
+    
+    message(sprintf("Completed annual metrics for %s/%s", which(site_ids == id), length(site_ids)))
+    
+    return(annual_metrics)
+  }) %>% 
+    bind_rows() %>% 
+    readr::write_csv(target_name)
+  
+}
+
+# Collection of functions to calculate annual thermal metrics
+# All accept the same main data input, `date` and `wtr`.
+
+# It is assumed that these functions only get passed one
+# years worth of data.
+
+#' @param date a vector (or one column) with the dates of class `Date`
+#' @param wtr a vector (or one column) with water temperature values in deg C
+
+# Some also have the following parameters:
+#' @param depth vector (or column) with depth in meters
+
+#' @description # of days, beginning October 30 of the previous year and running 
+#' through June of the given year, that surface water is <4 degrees C
+winter_dur_0_4 <- function(date, wtr) {
+  
+}
+
+#' @description Coefficient of Variation of surface temperature from 30-60 days post ice off.
+coef_var_30_60 <- function(date, wtr) {
+  # ice dates are in 4_inputs in sciencebase
+}
+
+#' @description Coefficient of Variation of surface temperature from 0-30 days post ice off.
+coef_var_0_30 <- function(date, wtr) {
+  
+}
+
+#' @description Julian date at which stratification sets up (longest stratified period)
+stratification_onset_yday <- function(date, wtr) {
+  
+}
+
+#' @description Duration of stratified period
+stratification_duration <- function(date, wtr) {
+  
+}
+
+#' @description avg. thermocline depth in stratified period
+sthermo_depth_mean <- function(date, wtr) {
+  
+}
+
+#' @description Maximum observed surface temperature
+peak_temp <- function(date, wtr, depth) {
+  max(wtr[which(depth == 0)], na.rm = TRUE)
+}
+
+#' @description Cumulative sum of degrees > 0 for entire year
+gdd_wtr_0c <- function(date, wtr) {
+  calc_gdd(wtr, 0)
```

Comment:
ahh, I see - you are calling a more generic GDD function after all. Seems "data" isn't needed as an input? I'd still also suggest you don't create these other wrapper functions. There may be a need to add more GDD base temperature outputs, and it would be better to avoid creating a new function each time.